nixos/modules/services/torrent/flood.nix

```
@@ -0,0 +1,115 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.services.flood;
+in
+{
+  meta.maintainers = with lib.maintainers; [ thiagokokada ];
+
+  options.services.flood = {
+    enable = lib.mkEnableOption "flood";
+    package = lib.mkPackageOption pkgs "flood" { };
+    openFirewall = lib.mkEnableOption "" // {
+      description = "Whether to automatically open the necessary ports in the firewall.";
+      default = true;
+    };
+    user = lib.mkOption {
+      type = lib.types.str;
+      description = "User account used to run flood.";
+      default = "flood";
+    };
+    group = lib.mkOption {
+      type = lib.types.str;
+      description = "Group account used to run flood.";
+      default = "flood";
+    };
+    port = lib.mkOption {
+      type = lib.types.int;
+      description = "Port to bind webserver.";
+      default = 3000;
+    };
+    host = lib.mkOption {
+      type = lib.types.str;
+      description = "Host to bind webserver.";
+      default = "0.0.0.0";
+    };
+    runDir = lib.mkOption {
+      type = lib.types.path;
+      description = "Directory to store Flood configuration. Needs to be readable/writable by Flood's user/group.";
+      default = "/var/lib/flood";
+    };
+    extraArgs = lib.mkOption {
+      type = with lib.types; listOf str;
+      description = "Extra arguments passed to `flood`.";
+      default = [ ];
+      example = [ "--baseuri=/" ];
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    users = {
+      users = lib.optionalAttrs (cfg.user == "flood") {
+        flood = {
+          group = cfg.group;
+          isSystemUser = true;
+        };
+      };
+      groups = lib.optionalAttrs (cfg.group == "flood") {
+        flood = { };
+      };
+    };
+
+    systemd.services.flood = {
+      description = "A web UI for rTorrent with a Node.js backend and React frontend.";
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+      unitConfig = {
+        Documentation = "https://github.com/jesec/flood/wiki";
+      };
+      serviceConfig = {
+        User = cfg.user;
+        Group = cfg.group;
+        Type = "simple";
+        KillMode = "process";
+        Restart = "on-failure";
+        RestartSec = "3s";
+        ExecStart = lib.concatStringsSep " " ([
+          "${lib.getExe cfg.package}"
+          "--host"
+          cfg.host
+          "--port"
+          (toString cfg.port)
+          "--rundir"
+          cfg.runDir
+        ] ++ cfg.extraArgs);
```

```suggestion
        ExecStart = utils.escapeSystemdExecArgs ([
          (lib.getExe cfg.package)
          "--host"
          cfg.host
          "--port"
          (toString cfg.port)
          "--rundir"
          cfg.runDir
        ] ++ cfg.extraArgs);
```

Make sure we properly escape the exec args.